### PR TITLE
Add new Slack and PagerDuty secret types

### PR DESCRIPTION
**Pull request by colin014**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
Add 2 new secret types: `slack` and `pagerduty`.

Slack fields
---

| Field name | Description
|--------|---
| apiUrl | Slack url to send alerts

PagerDuty fields
---

| Field name | Description
|--------|---
| integrationKey | The PagerDuty integration key

### Why?
In the `Monitoring` feature we want to use `slack` and `pagerduty` secrets.

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] OpenAPI and Postman files updated (if needed)

### CODE REVIEW

File: pkg/secret/secret.go [hunk 1]
```diff
@@ -343,6 +360,19 @@ var DefaultRules = map[string]Meta{
 			{Name: VaultToken, Required: true, Opaque: true, Description: "Token for Vault"},
 		},
 	},
+	SlackSecretType: {
+		Fields: []FieldMeta{
+			{Name: SlackApiUrl, Required: true, Opaque: true, Description: "Slack url to send alerts"},
```
Comment (sagikazarmark) [inline pkg/secret/secret.go]:
```suggestion
			{Name: SlackApiUrl, Required: true, Opaque: true, Description: "Slack URL to send alerts to"},
```

File: pkg/secret/secret.go [hunk 2]
```diff
@@ -343,6 +360,19 @@ var DefaultRules = map[string]Meta{
...
+			{Name: SlackChannel, Required: true, Opaque: true, Description: "Channel name for the alerts"},
```
Comment (sagikazarmark) [inline pkg/secret/secret.go]: Is this supposed to be in the secret? IMHO this is rather a feature setting. (Eg. use the same secret, but separate channels for separate clusters)

File: pkg/secret/secret.go [hunk 3]
```diff
@@ -343,6 +360,19 @@ var DefaultRules = map[string]Meta{
...
+		},
+	},
+	PagerDutySecretType: {
+		Fields: []FieldMeta{
+			{Name: PagerDutyUrl, Required: true, Opaque: true, Description: "PagerDuty service endpoint"},
+			{Name: PagerDutyRoutingKey, Required: true, Opaque: true, Description: "The PagerDuty integration key (when using PagerDuty integration type 'Events API v2')."},
```
Comment (sagikazarmark) [inline pkg/secret/secret.go]: Not sure if this should be in a secret either.

**Comment by colin014** [comment]
I updated the description and the PR on the basis of the discussions

**Pull request merged**